tests/ha/test_control_pod_restart.py

```diff
@@ -383,3 +401,286 @@ def test_failover_control_pod_kubectl(self):
+                                                    skipcleanup=True, nclients=50, nsamples=50)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 2: Performed WRITEs with variable sizes objects for parallel READs.")
+
+        LOGGER.info("Starting three independent background threads for READs, WRITEs & DELETEs.")
+        LOGGER.info("Step 3: Start Continuous DELETEs in background on random %s buckets",
+                    del_bucket)
+        bucket_list = list(s3_data.keys())
+        get_random_buck = self.system_random.sample(bucket_list, del_bucket)
+        args = {'test_prefix': test_prefix_del, 'test_dir_path': self.test_dir_path,
+                'skipput': True, 'skipget': True, 'bkt_list': get_random_buck, 'output': del_output}
+        thread_del = threading.Thread(target=self.ha_obj.put_get_delete,
+                                      args=(event, s3_test_obj,), kwargs=args)
+        thread_del.daemon = True  # Daemonize thread
+        thread_del.start()
+        LOGGER.info("Step 3: Successfully started DELETEs in background for %s buckets", del_bucket)
+
+        LOGGER.info("Step 4: Perform WRITEs with variable object sizes in background")
+        test_prefix_write = 'test-write-40370'
+        args = {'s3userinfo': list(users.values())[1], 'log_prefix': test_prefix_write,
+                'nclients': 5, 'nsamples': 50, 'skipread': True, 'skipcleanup': True,
+                'output': wr_output}
+        thread_wr = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        thread_wr.daemon = True  # Daemonize thread
+        thread_wr.start()
+        LOGGER.info("Step 4: Successfully started WRITEs with variable sizes objects in background")
+        time.sleep(HA_CFG["common_params"]["20sec_delay"])    # delay to allow s3bench installation
+
+        LOGGER.info("Step 5: Perform READs and verify DI on the written data in background")
+        output_rd = Queue()
+        args = {'s3userinfo': list(users.values())[1], 'log_prefix': test_prefix_read,
+                'nclients': 5, 'nsamples': 50, 'skipwrite': True, 'skipcleanup': True,
+                'setup_s3bench': False, 'output': output_rd}
+        thread_rd = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        thread_rd.daemon = True  # Daemonize thread
+        thread_rd.start()
+        LOGGER.info("Step 5: Successfully started READs and verified DI on the written data in "
+                    "background")
+
+        LOGGER.info("Control pod %s is hosted on %s node", self.control_pod_name, self.control_node)
+
+        failover_node = self.system_random.choice([ele for ele in self.host_worker_list if ele !=
+                                                   self.control_node])
+        LOGGER.debug("Fail over node is: %s", failover_node)
+
+        event.set()
```

Comment:
why do we need to set event as we would be expecting all IOs to be successful?

Reply:
Just to check if no failures are happening while pod is failing over. In case any IO fails, it will be easy to debug if it failed while pod failing over or after failover.